scripts/build.js

```diff
@@ -0,0 +1,110 @@
+const path = require('path');
+
+const { ucs2 } = require("punycode"); 
+const SimpleIcons = require('simple-icons');
+const svg2ttf = require('svg2ttf');
+const SVGPath = require('svgpath');
+const ttf2woff = require('ttf2woff');
+const ttf2woff2 = require('ttf2woff2');
+
+
+const DISTDIR = path.resolve(__dirname, '..', 'font');
+
+const buildSimpleIconsSvgFontFile = () => {
+  let fileContent = '<?xml version="1.0" standalone="no"?>\n'
+                  + '<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">\n'
+                  + '<svg xmlns="http://www.w3.org/2000/svg">\n'
+                  + '<metadata>CC0 1.0 Universal | simple-icons contributors</metadata>\n'
+                  + '<defs>\n'
+                  + '<font id="Simple Icons" horiz-adv-x="1200">\n'
+                  + '<font-face font-family="Simple Icons" units-per-em="1200" ascent="-1" descent="1200" />\n'
+                  + '<missing-glyph horiz-adv-x="0" />\n';
+  
+  let startUnicode = 0xea01;
+  let usedUnicodes = [];
+  
+  let unicodeHexBySlug = {}
+  
+  for (let iconTitle in SimpleIcons) {
+    let unicode = ucs2.decode(
+      String.fromCodePoint(startUnicode++)).map(
+        point => '&#x' + point.toString(16).toUpperCase() + ';')[0];
+    if (usedUnicodes.includes(unicode)) {
+      throw Error(`Unicodes must be unique. Found '${unicode}' repeated`)
+    }
+    
+    let icon = SimpleIcons[iconTitle];
+    const verticalTransformedPath = SVGPath(icon.path).scale(50, -50).toString();
+    
+    fileContent += `<glyph glyph-name="${icon.slug}" unicode="${unicode}" d="${verticalTransformedPath}" horiz-adv-x="1200" />\n`;
+    usedUnicodes.push(unicode);
+
+    unicodeHexBySlug[icon.slug] = {
+      unicode: unicode,
+      hex: icon.hex
+    };
+  }
+  fileContent += '</font>\n</defs>\n</svg>\n';
+  
+  const outputFilePath = path.join(DISTDIR, 'SimpleIcons.svg');
+  fs.writeFileSync(outputFilePath, fileContent);
+  console.log(`'${outputFilePath}' file built`);
+  
+  return unicodeHexBySlug;
+}
+
+const buildSimpleIconsCssFile = (unicodeHexBySlug) => {
+  let simpleIconsCss = fs.readFileSync(
+      path.resolve(__dirname, '..', 'preview', 'css', 'base.css'));
+  
+  for (let slug in unicodeHexBySlug) {
+    let icon = unicodeHexBySlug[slug];
+    
+    // &#xF26E; -> \f26e
+    const cssDecodedUnicode = icon.unicode.replace('&#x', '\\').replace(';', '').toLowerCase();
+    simpleIconsCss += '\n'
+                    +   `.simpleicons-${slug}::before { content: "${cssDecodedUnicode}"; }\n`
+                    +   `.simpleicons-${slug}.simpleicons--color::before { color: #${icon.hex}; }\n`;
+  }
+  
+  const outputFilePath = path.join(DISTDIR, 'SimpleIcons.css');
+  fs.writeFileSync(outputFilePath, simpleIconsCss);
+  console.log(`'${outputFilePath}' file built`);
+}
+
+const buildSimpleIconsTtfFontFile = () => {
+  const ttf = svg2ttf(fs.readFileSync(path.join(DISTDIR, 'SimpleIcons.svg'), 'utf8'), {});
```

Comment:
Would prefer not having to read this after it has been created and instead just pass the `SimpleIcons.svg` content as an argument.

Reply:
Sorry for the mess, I've defined now built assets filepaths as global variales and passed the contents of the files as you have requested. I hope that this could be more readable.

Reply:
No problem. It was quite readable but I do think it is much nicer now. The main reason I requested it is because reading it from disk is a bit of a waste of time if you already have it in memory anyway.